dask/array/tests/test_array_core.py

```diff
@@ -370,6 +370,229 @@ def test_concatenate_fixlen_strings():
               da.concatenate([a, b]))
 
 
+def test_block_simple_row_wise():
+        a_2d = np.ones((2, 2))
+        b_2d = 2 * a_2d
+        desired = np.array([[1, 1, 2, 2],
+                            [1, 1, 2, 2]], dtype=float)
+        result = da.block([a_2d, b_2d])
+        assert_eq(desired, result)
+
+
+def test_block_simple_column_wise():
+        a_2d = np.ones((2, 2))
+        b_2d = 2 * a_2d
+        expected = np.array([[1, 1],
+                             [1, 1],
+                             [2, 2],
+                             [2, 2]], dtype=float)
+        result = da.block([[a_2d], [b_2d]])
+        assert_eq(expected, result)
```

Comment:
Is it feasible to test against `np.block` instead?

Reply:
If we want to require NumPy 1.13, yes it is possible. Though it does seem useful to be able to test with older versions of NumPy. Suppose we could add some optional checks in addition to these that only run with NumPy 1.13+.

Reply:
We could require Numpy 1.13 just on these tests.  Heck, we might even only publish the block implementation if NumPy 1.13 was present.  

I don't care strongly either way and am happy to defer to your judgement here (you're more involved in this code and have done all of the work).  Generally I feel that we shouldn't be afraid to ask users to update when reasonable.

Reply:
My preference would be to add some checks against NumPy's `block` that are skipped if NumPy 1.13+ is not present.

Reply:
Skipping if numpy 1.13 isn't present and then using `np.block` sounds good to me as well